cl/people_db/filters.py

```diff
@@ -270,8 +286,21 @@ class Meta:
             "name": ALL_TEXT_LOOKUPS,
         }
 
+    def filter_roles(self, qs, name, value):
+        if not value:
+            return qs
+
+        role_filters = self.get_filters_for_join_table(name)
```

Comment:
I found that when passing an invalid filter like:

`?test__contains=Richard&filter_nested_results=true`
or
`?parties_represented__attorney_name=lorem&filter_nested_results=true`

Returns an error like:
Cannot resolve keyword 'attorney_name' into field. Choices are: attorney, attorney_id, date_action, docket, docket_id, id, party, party_id, role, role_raw

Or
`?attorney__test=23&filter_nested_results=true`

Unsupported lookup 'test' for ForeignKey or join on the field not permitted.

This is because the invalid filter is being passed to:
`queryset=Role.objects.filter(**role_filters)`

I believe this can be solved by verifying if the filters returned by `get_filters_for_join_table` are valid `Role` fields.

We could do something like: 
`valid_role_fields = set(field.name for field in Role._meta.get_fields())`

And use the returned fields to filter out filters returned by `get_filters_for_join_table`

So invalid filters passed could be just ignored?

Reply:
> So invalid filters passed could be just ignored?

Thanks for pointing that out, @albertisfu!

There's a more elegant way to handle invalid filters in [Django-filter](https://github.com/carltongibson/django-filter/tree/main). The [BaseFilter](https://github.com/carltongibson/django-filter/blob/2494df96c6387a9fa411fcb00b696b15dfd9216b/django_filters/filterset.py#L186) class utilizes a form for validation. We can leverage this to filter out any invalid filters before applying them to the queryset.

Here's a link to the relevant section of the `filter_queryset` method in django-filter:

https://github.com/carltongibson/django-filter/blob/2494df96c6387a9fa411fcb00b696b15dfd9216b/django_filters/filterset.py#L228-L237